src/Database/LSMTree/Internal/Snapshot.hs

```diff
@@ -301,6 +304,12 @@ openWriteBuffer ::
   -> WriteBufferFsPaths
   -> m (WriteBuffer, Ref (WriteBufferBlobs m h))
 openWriteBuffer reg resolve hfs hbio uc activeDir snapWriteBufferPaths = do
+  -- Check the checksums
+  (expectedChecksumForKOps, expectedChecksumForBlob) <-
+    CRC.expectValidFile (writeBufferChecksumsPath snapWriteBufferPaths) . fromChecksumsFileForWriteBufferFiles
+      =<< CRC.readChecksumsFile hfs (writeBufferChecksumsPath snapWriteBufferPaths)
+  checkCRC hfs hbio False (unForKOps expectedChecksumForKOps) (writeBufferKOpsPath snapWriteBufferPaths)
+  checkCRC hfs hbio False (unForBlob expectedChecksumForBlob) (writeBufferBlobPath snapWriteBufferPaths)
```

Comment:
Hmm, I just noticed that we're in fact reading the blob file twice: once to check the CRC, then once below to copy it from the snapshot directory to the active directory. Maybe leave a TODO that we could optimise this by computing the CRC and copying the file in one traversal of the file?

Reply:
Shouldn't the fs cache help with this?

Reply:
It will probably faster than if were to do actual I/O, but still a double traversal. And on more memory-constrained systems, we might not be able to benefit from the cache much